addons/survey/wizard/survey_invite.py

```diff
@@ -170,12 +171,24 @@ def _prepare_answers(self, partners, emails):
             if self.existing_mode == 'resend':
                 partners_done = existing_answers.mapped('partner_id')
                 emails_done = existing_answers.mapped('email')
-                answers |= existing_answers
 
+                # only add the last answer for each user of each type (partner_id & email)
+                # to have only one mail sent per user
+                for partner_done in partners_done:
+                    answers |= next(iter(existing_answers
+                        .filtered(lambda answer: answer.partner_id == partner_done)
+                        .sorted(lambda answer: answer.create_date, reverse=True)))
+
+                for email_done in emails_done:
+                    answers |= next(iter(existing_answers
+                        .filtered(lambda answer: answer.email == email_done)
+                        .sorted(lambda answer: answer.create_date, reverse=True)))
+
+        invite_token = str(uuid.uuid4())
```

Comment:
I would let the default value on field definition handle this (create a new pool case).

Reply:
Not necessary anymore (as discussed).